### PR TITLE
Autonick

### PR DESCRIPTION
**Pull request by Moth-Tolias**
hullo! i made this script a while ago and it's been useful to me, so i asked @BenLubar about it and xey said to submit a PR here.

### CODE REVIEW

File: autonick.lua
Comment (myk002) [inline autonick.lua:23]: This is a user-written configuration file, right? It should probably go in dfhack-config/autonick.txt

File: autonick.lua
Comment (myk002) [inline autonick.lua:8]: It would be useful to also accept comments (lines starting with a '#' character)

File: autonick.lua
Comment (Moth-Tolias) [reply autonick.lua:23]: quite right! i normally only use the default /data configs, so i totally overlooked dfhack-config.

File: autonick.lua
Comment (myk002) [inline autonick.lua:28]: Consider table.insert(names, line) so you don't have to manually track index

File: autonick.lua
Comment (myk002) [inline autonick.lua:27]: line = line:trim() would be useful here, to remove leading and trailing whitespace

File: autonick.lua
Comment (Moth-Tolias) [reply autonick.lua]: still need to write the default file, but the location has been changed now.

File: autonick.lua
Comment (myk002) [inline autonick.lua:47]: You can reduce complexity here by breaking out of the loop if #names == 0:
if #names == 0 then break end
if dfhack.units.isCitizen(...

File: autonick.lua
Comment (myk002) [inline autonick.lua:39]: Instead of looping through to remove taken nicknames, how about filtering them out when we build the names list?

You could add this before the io.lines loop above (sorry for the formatting.. I'm writing this on my phone):

local taken = {}
for _,unit in ipairs(df.global.world.units.active) do
    if dfhack.units.isCitizen(unit) and
    unit.name.nickname ~= "" then taken[unit.name.nickname] = true end
end

And then when building names, check if taken[name] is truthy

File: autonick.lua
Comment (myk002) [inline autonick.lua:6]: This might be a bit clearer:

Gives dwarves unique nicknames chosen randomly from ``dfhack-config/autonick.txt``.

File: autonick.lua
Comment (myk002) [inline autonick.lua:27]: it would be better to assign the trimmed value back to line -- otherwise the space-padded version will get inserted into the names list.

also, line:startswith('#') also works and is shorter

**Comment by myk002** [comment]
and an overall comment - there isn't a strict code style standard, but DFHack generally doesn't use semicolons for lua scripts

File: autonick.lua
Comment (Moth-Tolias) [reply autonick.lua]: will do this next.

**Comment by Moth-Tolias** [comment]
should be all good now =] i'm not sure where i would submit the example autonick.txt file though - the main df hack once [if] this gets merged?

**Review by myk002 — APPROVED** [review]
@lethosor has the final say, but this looks good to me! Thanks for implementing all the suggestions!

**Comment by myk002** [comment]
> should be all good now =] i'm not sure where i would submit the example autonick.txt file though - the main df hack once [if] this gets merged?

Yeah, once this is merged, add the default autonick.txt to https://github.com/DFHack/dfhack/tree/develop/dfhack-config. That gets installed to <df_dir>/dfhack-config/default/ and anything in default gets copied to <df_dir>/dfhack-config/ if the corresponding fine doesn't already exist

File: autonick.lua
Comment (myk002) [inline autonick.lua:9]: I just noticed that repeat entries are *not* discarded. They could be, though, if we add the names read from the config file to the `taken` set as we read them. Then any repeat entries will be properly ignored.

maybe rename `taken` to `seen` as well to reflect its new role.

**Review by lethosor — APPROVED** [review]

**Comment by Moth-Tolias** [comment]
thanks for merging! i'm really glad this got in =]

**Pull request merged**
